app/models/prison.rb

```diff
@@ -1,32 +1,62 @@
 class Prison
+  attr_accessor :address, :adult_age, :booking_window, :canned_responses,
+    :email, :enabled, :finder_slug, :lead_days,
+    :name, :nomis_id, :phone, :reason,
+    :slot_anomalies, :slots, :unbookable,
+    :works_weekends
+
   DEFAULT_LEAD_DAYS = 3.freeze
   DEFAULT_BOOKING_WINDOW = 28.freeze
 
   class PrisonNotFound < StandardError; end
 
-  def self.find(prison_name)
-    prison_hash = Rails.configuration.prison_data[prison_name]
-    if prison_hash.nil?
+  def self.find(prison_name, raise_error = false)
+    prison = all.detect{ |p| p.name == prison_name }
+    if raise_error && prison.nil?
       raise PrisonNotFound, "Can't find prison #{prison_name.inspect}"
     end
-    new(prison_name, prison_hash)
+    prison
+  end
+
+  def initialize(opts = {})
+    opts.with_indifferent_access.map do |k, v|
+      instance_variable_set("@#{k}", v)
+    end
+    @booking_window = DEFAULT_BOOKING_WINDOW if @booking_window.blank?
+  end
+
+  def self.create(opts = {})
+    prison = new(opts)
+    Rails.configuration.prison_data << prison
+    prison
+  end
+
+  def self.all
+    Rails.configuration.prison_data
   end
 
-  def initialize(name, prison_hash)
-    @name = name
-    @prison_hash = prison_hash.with_indifferent_access
+  def self.enabled
+    all.select{ |p| p.enabled == true }
   end
 
-  attr_reader :name
+  def self.names
+    all.map(&:name).sort
+  end
 
-  delegate :fetch, to: :@prison_hash
+  def self.nomis_ids
+    all.map(&:nomis_id).reject(&:nil?).sort
+  end
+
+  def enabled?
+    enabled
+  end
 
   def unbookable_dates
-    (fetch(:unbookable) { Array.new }).to_set
+    (unbookable || Array.new).to_set
   end
 
   def visiting_slots
-    fetch(:slots)
+    @visiting_slots ||= slots
```

Comment:
if this method is now just an alias, why memoise. (Or, why memoise here, and not there?)